code/game/mecha/mecha.dm

```diff
@@ -945,10 +942,29 @@
 		else
 			step(obstacle,src.dir)
 
+	else if(istype(obstacle, /turf))

```

Comment:
Should combine this with the phasing checks so that a non-phasing mech still calls `..()` at the end of the chain.